src/mantarray_file_manager/files.py

```diff
@@ -445,17 +453,53 @@ def get_raw_reference_reading(self) -> NDArray[(2, Any), int]:
                 self.get_reference_sampling_period_microseconds()
                 / MICROSECONDS_PER_CENTIMILLISECOND
             )
+
             ref_data = self._h5_file["reference_sensor_readings"]
 
             times = np.arange(len(ref_data), dtype=np.int32) * time_step
             len_time = len(times)
 
+            new_time_delta = self._check_for_trimmed_file(
+                times, time_delta_centimilliseconds
+            )
+
+            if new_time_delta != 0:
+                time_delta_centimilliseconds = new_time_delta
+
             self._raw_ref_reading = np.array(
                 (times + time_delta_centimilliseconds, ref_data[:len_time]),
                 dtype=np.int32,
             )
+
         return self._raw_ref_reading
 
+    def _check_for_trimmed_file(
+        self, times: NDArray[(1, Any), int], time_delta_centimilliseconds: int
+    ) -> int:
+        file = self.get_h5_file()
+        metadata_keys = set(file.attrs.keys())
+        if str(IS_FILE_ORIGINAL_UNTRIMMED_UUID) in metadata_keys:
```

Comment:
for future: Python is built around the philospohy of "easier to ask forgiveness than permission", as opposed to some other languages (which I think includes Java) which are built around "look before you leap".  So since we already have a method to look up a metadata by the UUID, you can just call that and catch an exception if it gets thrown, rather than all of these `if` checks to see if the metadata key exists or not

https://devblogs.microsoft.com/python/idiomatic-python-eafp-versus-lbyl/